pkg/inject/inject.go

```diff
@@ -110,6 +101,18 @@ type ResourceConfig struct {
 	}
 }
 
+// Workload contains the information for a given conf
+type Workload struct {
```

Comment:
As we can check the `kind` directly by 
```go
//IsWorkloadNamespace checks if a given config is of type namespace
func (conf *ResourceConfig) IsWorkloadNamespace() bool {
	return strings.ToLower(conf.workload.metaType.Kind) == k8s.Namespace
}
```

I don't see a point in making `workload` public and taking all this hassle.